### DISCUSSION TITLE
GDScript Modulo (%) with Floats: Warn user to use fmod() instead

### DISCUSSION THREAD

**Discussion by dotRelith** [godotengine/godot-proposals#9996] [Scripting]
Issue:

The modulo operator (%) in GDScript currently operates only on integers. When used with floating-point operands, it throws the given error: `Invalid operands to operator %, float and float.` 

Proposed Solution:

To perform modulo operations on floating-point numbers in GDScript, the only way I know of is to use the fmod() function. I spent 30 minutes looking for a solution when I could've used fmod() instead. It would be helpful if the script language could warn the user about this limitation and suggest using fmod() function.

**Comment by Calinou** [comment]
- Pull request opened: https://github.com/godotengine/godot/pull/93328

**Comment by Mickeon** [comment]
One time on RocketChat I asked why `%` does not work for floats in **GDScript** and the answer was... for legacy reasons, and that's mostly it, really. Adding support for it would be relatively simple, but I guess one may argue it is unexpected, as it does not work for most other programming languages.

**Reply by dalexeev** [reply]
Given the situation with the confusing division operator (integer vs real division), perhaps it's good for float modulo to have a global function instead of an operator. Although here confusion is less likely, unlike division.

**Comment by Cerilam** [comment]
The issue is still present in godot 4.4. 
It's a pretty dumb and easy to fix problem to block the user for x minutes instead of printing a simple "use fmod() instead"

**Reply by Mickeon** [reply]
Alright, since it's so easy to fix, someone should feel free to make a PR about it so it can be reviewed

**Reply by Cerilam** [reply]
Ho yeah my message was pretty lame, sorry about that. 
It wasn't meant to be malicious.
I'll check if I can make a PR for it

**Reply by Calinou** [reply]
There's already an open PR, as commented above. I rebased it yesterday and addressed review comments.
